### PR TITLE
Set NFSv4 client ID before mount

### PR DESCRIPTION
**Pull request by alvarocarvajald**
Commits baff123cb6210f8b959b421ffeb57c6adc33c530 and be30deb95fe7e6654fc584b81632e3f1c688973f introduced the configuration of  an unique NFS client ID while copying SAP installation media from a NFS share. This configuration was done as defined in those commits after the share had been mounted, however there still exists a race condition when more than one system identified with the same NFS client ID attempts to mount the share.
    
This commit introduces changes which load the `nfs` kernel module before the `mount` command, supplying to it the `nfs4_unique_id` parameter on load so the client has an unique ID before any NFS operation, including before the `mount` command which can be blocked if there are client ID clashes.
    
This commit also replaces the use of a sha256 for an md5 digest, as the sha256 string is too long for `nfs4_unique_id` and caused `modprobe` to fail. A sha256 digest is good enough for `/sys/fs/nfs/net/nfs_client/identifier`, but as described on 61c9fc6e09838bfa941083d8e38a70edf4e42bcc, this path is not available on kernels older than 5.3, so we are instead using the `nfs4_unique_id` parameter from the `nfs` kernel module.

- Related ticket: https://progress.opensuse.org/issues/135923
- Needles: N/A
- Verification run: [15-SP1](https://openqaworker15.qa.suse.cz/tests/231985), [12-SP5](https://openqaworker15.qa.suse.cz/tests/231986), [15-SP5](https://openqaworker15.qa.suse.cz/tests/231987), [15-SP5 w/NFS_TIMEO=50](http://openqaworker15.qa.suse.cz/tests/232138)

### CODE REVIEW

File: lib/sles4sap.pm [hunk 1]
```diff
@@ -396,16 +396,18 @@ sub copy_media {
     if ($proto eq 'nfs') {
         my $nfs_timeo = get_var('NFS_TIMEO');
         $options = $nfs_timeo ? "timeo=$nfs_timeo,rsize=16384,wsize=16384,ro" : 'rsize=16384,wsize=16384,ro';
+        # Attempt to force a unique NFSv4 client id
+        assert_script_run 'modprobe nfs';
+        assert_script_run 'while (! ls /sys/module/nfs/parameters/nfs4_unique_id); do sleep 1; done';    # Wait for file to be available
```
Comment (okurz) [inline lib/sles4sap.pm:401]:
```suggestion
        assert_script_run 'until ls /sys/module/nfs/parameters/nfs4_unique_id; do sleep 1; done';    # Wait for file to be available
```

**Comment by alvarocarvajald** [comment]
Added *WIP*.

Writing on `/sys/module/nfs/parameters/nfs4_unique_id` before the mount fails: https://openqaworker15.qa.suse.cz/tests/231962#step/hana_install/39

However, it seems it was failing before (after the mount command) as well: https://openqaworker15.qa.suse.cz/tests/231910#step/hana_install/42

Saw jobs in openqa.suse.de where this fails and passes probably depending on the qcow2 image used in the test. There are [tests](https://openqa.suse.de/tests/12224958#step/hana_install/27) where writing on the file succeeds, but still lead to a failure later.

File: lib/sles4sap.pm (hunk 1 above)
Comment (alvarocarvajald) [reply lib/sles4sap.pm]: Thanks. Changing this.

**Comment by alvarocarvajald** [comment]
sha256sum was too long for `nfs4_unique_id`, so I changed it for `md5sum` and it's working. See commit message or pull request description for more details. Removing WIP label.

**Review by okurz — APPROVED** [review]

**Review by a-kpappas — APPROVED** [review]

**Pull request merged**